streams/src/main/java/org/apache/kafka/streams/processor/assignment/TaskAssignmentUtils.java

```diff
@@ -407,4 +543,345 @@ private static boolean hasValidRackInformation(final TaskInfo task,
         }
         return true;
     }
+
+    private static Map<ProcessId, KafkaStreamsAssignment> tagBasedStandbyTaskAssignment(final ApplicationState applicationState,
+                                                                                        final Map<ProcessId, KafkaStreamsAssignment> kafkaStreamsAssignments) {
+        final int numStandbyReplicas = applicationState.assignmentConfigs().numStandbyReplicas();
+        final Map<TaskId, Integer> tasksToRemainingStandbys = applicationState.allTasks().values().stream()
+            .collect(Collectors.toMap(TaskInfo::id, taskInfo -> numStandbyReplicas));
+        final Map<ProcessId, KafkaStreamsState> streamStates = applicationState.kafkaStreamsStates(false);
+
+        final Set<String> rackAwareAssignmentTags = new HashSet<>(getRackAwareAssignmentTags(applicationState));
+        final TagStatistics tagStatistics = new TagStatistics(applicationState);
+
+        final ConstrainedPrioritySet standbyTaskClientsByTaskLoad = standbyTaskPriorityListByLoad(streamStates, kafkaStreamsAssignments);
+
+        final Set<TaskId> statefulTaskIds = applicationState.allTasks().values().stream()
+            .filter(TaskInfo::isStateful)
+            .map(TaskInfo::id)
+            .collect(Collectors.toSet());
+        final Map<UUID, KafkaStreamsAssignment> clientsByUuid = kafkaStreamsAssignments.entrySet().stream().collect(Collectors.toMap(
+            entry -> entry.getKey().id(),
+            Map.Entry::getValue
+        ));
+        final Function<KafkaStreamsState, Map<String, String>> clientTagGetter = createClientTagGetter(applicationState);
+
+        final Map<TaskId, ProcessId> pendingStandbyTasksToClientId = new HashMap<>();
+        for (final TaskId statefulTaskId : statefulTaskIds) {
+            for (final KafkaStreamsAssignment assignment : clientsByUuid.values()) {
+                if (assignment.tasks().containsKey(statefulTaskId)) {
```

Comment:
one more minor bug: in the original code this is actually checking for assigned _active_ tasks specifically, so this should be 
```suggestion
                if (assignment.tasks().containsKey(statefulTaskId) && assignment.tasks().get(statefulTaskId).type() == Type.ACTIVE) {
```

Reply:
@apourchet make sure to address this in the followup PR